### ISSUE TITLE
[Bug] example/with_flutter doesn't compile

### ISSUE THREAD

**Issue by patmuk** [fzyzcjy/flutter_rust_bridge#1433]
### Describe the bug

tl;dr:

```
--- a/frb_example/with_flutter/pubspec.yaml
+++ b/frb_example/with_flutter/pubspec.yaml
@@ -53,7 +53,7 @@ dev_dependencies:
-  ffigen: ^8.0.0
+  ffigen: ^9.0.0
```
Solves the dependency issue.   
I recommend applying this patch, even though I saw that updating to 10.0 is on the roadmap. 
Thus, newcomers are not confused that the example doesn't work.

Please read on, as there is another issue after fixing the dependency issue!

Longer version:  
In Master, when checking out the example/with_flutter, running '''flutter run''' leads to dependency issues (on an Intel Mac - but that should not matter):

```
patmuk@SanPats-mini ~/c/o/s/examples> git clone https://github.com/fzyzcjy/flutter_rust_bridge.git
Cloning into 'flutter_rust_bridge'...
remote: Enumerating objects: 114783, done.
remote: Counting objects: 100% (8175/8175), done.
remote: Compressing objects: 100% (2119/2119), done.
remote: Total 114783 (delta 5840), reused 8084 (delta 5751), pack-reused 106608
Receiving objects: 100% (114783/114783), 40.32 MiB | 2.21 MiB/s, done.
Resolving deltas: 100% (85099/85099), done.
patmuk@SanPats-mini ~/c/o/s/examples> cd flutter_rust_bridge/
patmuk@SanPats-mini ~/c/o/s/e/flutter_rust_bridge (master)> cd frb_example/
patmuk@SanPats-mini ~/c/o/s/e/f/frb_example (master)> cd with_flutter/
patmuk@SanPats-mini ~/c/o/s/e/f/f/with_flutter (master)> flutter run
Resolving dependencies... (2.0s)
Note: file is pinned to version 7.0.0 by integration_test from the flutter SDK.
See https://dart.dev/go/sdk-version-pinning for details.


Because ffigen >=2.0.0 <9.0.0 depends on file ^6.0.0 and every version of integration_test from sdk depends on file 7.0.0, ffigen >=2.0.0 <9.0.0 is incompatible with integration_test from sdk.
So, because flutter_rust_bridge_example depends on both integration_test from sdk and ffigen ^8.0.0, version solving failed.


You can try the following suggestion to make the pubspec resolve:
* Try upgrading your constraint on ffigen: flutter pub add dev:ffigen:^10.0.0
```

## second issue - Xcode settings are wrong
tl;dr:
The static library is configured wrongly:
```
patmuk@SanPats-mini ~/c/o/s/e/f/f/with_flutter (master)> flutter run
Resolving dependencies... 
(...)
Removing script build phase dependency analysis.
Adding input path to Thin Binary build phase.
Running pod install...                                           1,494ms
Running Xcode build...                                                  
 └─Compiling, linking and signing...                      2,645ms
Xcode build done.                                           23.6s
Failed to build iOS app
Error (Xcode): Library 'flutter_rust_bridge_example' not found


Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)


Could not build the application for the simulator.
Error launching application on iPhone 15 Pro.
```
After fixing this, the linker binary path is wrong:
```
Error (Xcode): linking with `/usr/bin/ld` failed: exit status: 1
Error (Xcode): could not compile `num-traits` (build script) due to previous error
Error (Xcode): could not compile `memoffset` (build script) due to previous error
Error (Xcode): could not compile `crossbeam-epoch` (build script) due to previous error
Error (Xcode): could not compile `crossbeam-utils` (build script) due to previous error
Error (Xcode): could not compile `proc-macro2` (build script) due to previous error
Error (Xcode): could not compile `libc` (build script) due to previous error

Error (Xcode): cargo build failed

Could not build the application for the simulator.
Error launching application on iPhone 15 Pro.
```
No changes I did to the Xcode setup solved these. This is the diff of the changes I made:

```
patmuk@SanPats-mini ~/c/o/s/e/f/f/with_flutter (master) [1]> git diff ios/Runner.xcodeproj/project.pbxproj 
diff --git a/frb_example/with_flutter/ios/Runner.xcodeproj/project.pbxproj b/frb_example/with_flutter/ios/Runner.xcodeproj/project.pbxproj
index a92ca3003..328a67bd7 100644
--- a/frb_example/with_flutter/ios/Runner.xcodeproj/project.pbxproj
+++ b/frb_example/with_flutter/ios/Runner.xcodeproj/project.pbxproj
@@ -3,10 +3,11 @@
        archiveVersion = 1;
        classes = {
        };
-       objectVersion = 51;
+       objectVersion = 54;
        objects = {
 
 /* Begin PBXBuildFile section */
+               108A3D4C2B27249500FDD921 /* libflutter_rust_bridge_example_static.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 108A3D4B2B27249500FDD921 /* libflutter_rust_bridge_example_static.a */; };
                1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
                3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
                74858FAF1ED2DC5600515810 /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 74858FAE1ED2DC5600515810 /* AppDelegate.swift */; };
@@ -14,10 +15,47 @@
                97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
                97C147011CF9000F007C117D /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FF1CF9000F007C117D /* LaunchScreen.storyboard */; };
                C48D7477B661DF2833A3FE76 /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 4BB7C2D62FEC8F24D046EA58 /* Pods_Runner.framework */; };
-               E2816725270C91AB00999B57 /* libflutter_rust_bridge_example.a in Frameworks */ = {isa = PBXBuildFile; fileRef = E2816724270C91AB00999B57 /* libflutter_rust_bridge_example.a */; };
                E2816729270DA93400999B57 /* bridge_generated.h in Resources */ = {isa = PBXBuildFile; fileRef = E2816728270DA93400999B57 /* bridge_generated.h */; };
 /* End PBXBuildFile section */
 
+/* Begin PBXContainerItemProxy section */
+               108A3D502B2724D800FDD921 /* PBXContainerItemProxy */ = {
+                       isa = PBXContainerItemProxy;
+                       containerPortal = 108A3D4D2B2724D800FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                       proxyType = 1;
+                       remoteGlobalIDString = CA01CD485C98D7C61D2221EA;
+                       remoteInfo = "flutter_rust_bridge_example-staticlib";
+               };
+               108A3D542B2724D800FDD921 /* PBXContainerItemProxy */ = {
+                       isa = PBXContainerItemProxy;
+                       containerPortal = 108A3D4D2B2724D800FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                       proxyType = 2;
+                       remoteGlobalIDString = CA00CF89B15D2B3BABD1B908;
+                       remoteInfo = "flutter_rust_bridge_example-cdylib";
+               };
+               108A3D562B2724D800FDD921 /* PBXContainerItemProxy */ = {
+                       isa = PBXContainerItemProxy;
+                       containerPortal = 108A3D4D2B2724D800FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                       proxyType = 2;
+                       remoteGlobalIDString = CA01CD485C98AD1E3DC02A5C;
+                       remoteInfo = "flutter_rust_bridge_example-staticlib";
+               };
+               108A3D642B2726E900FDD921 /* PBXContainerItemProxy */ = {
+                       isa = PBXContainerItemProxy;
+                       containerPortal = 108A3D412B271D8600FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                       proxyType = 2;
+                       remoteGlobalIDString = CA00CF89B15D2B3BABD1B908;
+                       remoteInfo = "flutter_rust_bridge_example-cdylib";
+               };
+               108A3D662B2726E900FDD921 /* PBXContainerItemProxy */ = {
+                       isa = PBXContainerItemProxy;
+                       containerPortal = 108A3D412B271D8600FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                       proxyType = 2;
+                       remoteGlobalIDString = CA01CD485C98AD1E3DC02A5C;
+                       remoteInfo = "flutter_rust_bridge_example-staticlib";
+               };
+/* End PBXContainerItemProxy section */
+
 /* Begin PBXCopyFilesBuildPhase section */
                9705A1C41CF9048500538489 /* Embed Frameworks */ = {
                        isa = PBXCopyFilesBuildPhase;
@@ -32,6 +70,9 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
+               108A3D412B271D8600FDD921 /* flutter_rust_bridge_example.xcodeproj */ = {isa = PBXFileReference; lastKnownFileType = "wrapper.pb-project"; name = flutter_rust_bridge_example.xcodeproj; path = ../../rust/flutter_rust_bridge_example
.xcodeproj; sourceTree = "<group>"; };
+               108A3D4B2B27249500FDD921 /* libflutter_rust_bridge_example_static.a */ = {isa = PBXFileReference; explicitFileType = archive.ar; path = libflutter_rust_bridge_example_static.a; sourceTree = BUILT_PRODUCTS_DIR; };
+               108A3D4D2B2724D800FDD921 /* flutter_rust_bridge_example.xcodeproj */ = {isa = PBXFileReference; lastKnownFileType = "wrapper.pb-project"; name = flutter_rust_bridge_example.xcodeproj; path = /Users/patmuk/code/own/sherry/examples
/flutter_rust_bridge/frb_example/with_flutter/rust/flutter_rust_bridge_example.xcodeproj; sourceTree = "<absolute>"; };
                1498D2321E8E86230040F4C2 /* GeneratedPluginRegistrant.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = GeneratedPluginRegistrant.h; sourceTree = "<group>"; };
                1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = GeneratedPluginRegistrant.m; sourceTree = "<group>"; };
                3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.xml; name = AppFrameworkInfo.plist; path = Flutter/AppFrameworkInfo.plist; sourceTree = "<group>"; 
};
@@ -58,7 +99,7 @@
                        isa = PBXFrameworksBuildPhase;
                        buildActionMask = 2147483647;
                        files = (
-                               E2816725270C91AB00999B57 /* libflutter_rust_bridge_example.a in Frameworks */,
+                               108A3D4C2B27249500FDD921 /* libflutter_rust_bridge_example_static.a in Frameworks */,
                                C48D7477B661DF2833A3FE76 /* Pods_Runner.framework in Frameworks */,
                        );
                        runOnlyForDeploymentPostprocessing = 0;
@@ -66,6 +107,24 @@
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
+               108A3D4E2B2724D800FDD921 /* Products */ = {
+                       isa = PBXGroup;
+                       children = (
+                               108A3D552B2724D800FDD921 /* flutter_rust_bridge_example.dylib */,
+                               108A3D572B2724D800FDD921 /* libflutter_rust_bridge_example_static.a */,
+                       );
+                       name = Products;
+                       sourceTree = "<group>";
+               };
+               108A3D5F2B2726E900FDD921 /* Products */ = {
+                       isa = PBXGroup;
+                       children = (
+                               108A3D652B2726E900FDD921 /* flutter_rust_bridge_example.dylib */,
+                               108A3D672B2726E900FDD921 /* libflutter_rust_bridge_example_static.a */,
+                       );
+                       name = Products;
+                       sourceTree = "<group>";
+               };
                29B565F548BBDB2913D56D06 /* Pods */ = {
                        isa = PBXGroup;
                        children = (
@@ -96,6 +155,7 @@
                                97C146EF1CF9000F007C117D /* Products */,
                                E2816723270C91AB00999B57 /* Frameworks */,
                                29B565F548BBDB2913D56D06 /* Pods */,
+                               108A3D4D2B2724D800FDD921 /* flutter_rust_bridge_example.xcodeproj */,
                        );
                        sourceTree = "<group>";
                };
@@ -110,6 +170,7 @@
                97C146F01CF9000F007C117D /* Runner */ = {
                        isa = PBXGroup;
                        children = (
+                               108A3D412B271D8600FDD921 /* flutter_rust_bridge_example.xcodeproj */,
                                97C146FA1CF9000F007C117D /* Main.storyboard */,
                                97C146FD1CF9000F007C117D /* Assets.xcassets */,
                                97C146FF1CF9000F007C117D /* LaunchScreen.storyboard */,
@@ -125,6 +186,7 @@
                E2816723270C91AB00999B57 /* Frameworks */ = {
                        isa = PBXGroup;
                        children = (
+                               108A3D4B2B27249500FDD921 /* libflutter_rust_bridge_example_static.a */,
                                E2816724270C91AB00999B57 /* libflutter_rust_bridge_example.a */,
                                4BB7C2D62FEC8F24D046EA58 /* Pods_Runner.framework */,
                        );
@@ -150,6 +212,7 @@
                        buildRules = (
                        );
                        dependencies = (
+                               108A3D512B2724D800FDD921 /* PBXTargetDependency */,
                        );
                        name = Runner;
                        productName = Runner;
@@ -162,7 +225,7 @@
                97C146E61CF9000F007C117D /* Project object */ = {
                        isa = PBXProject;
                        attributes = {
-                               LastUpgradeCheck = 1300;
+                               LastUpgradeCheck = 1430;
                                ORGANIZATIONNAME = "";
                                TargetAttributes = {
                                        97C146ED1CF9000F007C117D = {
@@ -182,6 +245,16 @@
                        mainGroup = 97C146E51CF9000F007C117D;
                        productRefGroup = 97C146EF1CF9000F007C117D /* Products */;
                        projectDirPath = "";
+                       projectReferences = (
+                               {
+                                       ProductGroup = 108A3D4E2B2724D800FDD921 /* Products */;
+                                       ProjectRef = 108A3D4D2B2724D800FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                               },
+                               {
+                                       ProductGroup = 108A3D5F2B2726E900FDD921 /* Products */;
+                                       ProjectRef = 108A3D412B271D8600FDD921 /* flutter_rust_bridge_example.xcodeproj */;
+                               },
+                       );
                        projectRoot = "";
                        targets = (
                                97C146ED1CF9000F007C117D /* Runner */,
@@ -189,6 +262,37 @@
                };
 /* End PBXProject section */
 
+/* Begin PBXReferenceProxy section */
+               108A3D552B2724D800FDD921 /* flutter_rust_bridge_example.dylib */ = {
+                       isa = PBXReferenceProxy;
+                       fileType = "compiled.mach-o.dylib";
+                       path = flutter_rust_bridge_example.dylib;
+                       remoteRef = 108A3D542B2724D800FDD921 /* PBXContainerItemProxy */;
+                       sourceTree = BUILT_PRODUCTS_DIR;
+               };
+               108A3D572B2724D800FDD921 /* libflutter_rust_bridge_example_static.a */ = {
+                       isa = PBXReferenceProxy;
+                       fileType = archive.ar;
+                       path = libflutter_rust_bridge_example_static.a;
+                       remoteRef = 108A3D562B2724D800FDD921 /* PBXContainerItemProxy */;
+                       sourceTree = BUILT_PRODUCTS_DIR;
+               };
+               108A3D652B2726E900FDD921 /* flutter_rust_bridge_example.dylib */ = {
+                       isa = PBXReferenceProxy;
+                       fileType = "compiled.mach-o.dylib";
+                       path = flutter_rust_bridge_example.dylib;
+                       remoteRef = 108A3D642B2726E900FDD921 /* PBXContainerItemProxy */;
+                       sourceTree = BUILT_PRODUCTS_DIR;
+               };
+               108A3D672B2726E900FDD921 /* libflutter_rust_bridge_example_static.a */ = {
+                       isa = PBXReferenceProxy;
+                       fileType = archive.ar;
+                       path = libflutter_rust_bridge_example_static.a;
+                       remoteRef = 108A3D662B2726E900FDD921 /* PBXContainerItemProxy */;
+                       sourceTree = BUILT_PRODUCTS_DIR;
+               };
+/* End PBXReferenceProxy section */
+
 /* Begin PBXResourcesBuildPhase section */
                97C146EC1CF9000F007C117D /* Resources */ = {
                        isa = PBXResourcesBuildPhase;
@@ -207,10 +311,12 @@
 /* Begin PBXShellScriptBuildPhase section */
                3B06AD1E1E4923F5004D2608 /* Thin Binary */ = {
                        isa = PBXShellScriptBuildPhase;
+                       alwaysOutOfDate = 1;
                        buildActionMask = 2147483647;
                        files = (
                        );
                        inputPaths = (
+                               "${TARGET_BUILD_DIR}/${INFOPLIST_PATH}",
                        );
                        name = "Thin Binary";
                        outputPaths = (
@@ -221,6 +327,7 @@
                };
                9740EEB61CF901F6004384FC /* Run Script */ = {
                        isa = PBXShellScriptBuildPhase;
+                       alwaysOutOfDate = 1;
                        buildActionMask = 2147483647;
                        files = (
                        );
@@ -286,6 +393,14 @@
                };
 /* End PBXSourcesBuildPhase section */
 
+/* Begin PBXTargetDependency section */
+               108A3D512B2724D800FDD921 /* PBXTargetDependency */ = {
+                       isa = PBXTargetDependency;
+                       name = "flutter_rust_bridge_example-staticlib";
+                       targetProxy = 108A3D502B2724D800FDD921 /* PBXContainerItemProxy */;
+               };
+/* End PBXTargetDependency section */
+
 /* Begin PBXVariantGroup section */
                97C146FA1CF9000F007C117D /* Main.storyboard */ = {
                        isa = PBXVariantGroup;
@@ -363,6 +478,7 @@
                                ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
                                CLANG_ENABLE_MODULES = YES;
                                CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
+                               DYLIB_INSTALL_NAME_BASE = "@executable_path/../Frameworks/";
                                ENABLE_BITCODE = NO;
                                "EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "i386 arm64";
                                INFOPLIST_FILE = Runner/Info.plist;
@@ -376,7 +492,7 @@
                                );
                                PRODUCT_BUNDLE_IDENTIFIER = com.example.frbExample;
                                PRODUCT_NAME = "$(TARGET_NAME)";
-                               STRIP_STYLE ="non-global";
+                               STRIP_STYLE = "non-global";
                                SWIFT_OBJC_BRIDGING_HEADER = "Runner/Runner-Bridging-Header.h";
                                SWIFT_VERSION = 5.0;
                                VERSIONING_SYSTEM = "apple-generic";
@@ -497,6 +613,7 @@
                                ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
                                CLANG_ENABLE_MODULES = YES;
                                CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
+                               DYLIB_INSTALL_NAME_BASE = "@executable_path/../Frameworks/";
                                ENABLE_BITCODE = NO;
                                "EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "i386 arm64";
                                INFOPLIST_FILE = Runner/Info.plist;
@@ -525,6 +642,7 @@
                                ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
                                CLANG_ENABLE_MODULES = YES;
                                CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
+                               DYLIB_INSTALL_NAME_BASE = "@executable_path/../Frameworks/";
                                ENABLE_BITCODE = NO;
                                "EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "i386 arm64";
                                INFOPLIST_FILE = Runner/Info.plist;
```

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
same dependency problem when deleting generated files and running 'just _generate_bridge_with_flutter'
```

### OS

MacOS 14.1.2

### Version of `flutter_rust_bridge_codegen`

1.82.4f

**Comment by fzyzcjy** [comment]
Hi, thanks for the information!

I am working on V2 (hopefully release in a week) which changes a lot of the examples and how (by default) a project integrates Rust with Dart/Flutter. Therefore, one way may be to leave this issue open (such that everyone within this week can benefit from the information), and after V2 is there, check whether the problem is solved.

**Comment by patmuk** [comment]
works in v2.0.0-dev.2!
